app/main/views/send.py

```diff
@@ -1009,9 +1009,17 @@ def _send_notification(service_id, template_id):
 
     keys = []
     values = []
+    # Guarantee that the real phone number comes last, because some
+    # users will have placeholders like "add your second phone number"
+    # or something like as custom placeholders.
     for k, v in session["placeholders"].items():
-        keys.append(k)
-        values.append(v)
+        if k != "phone number":
+            keys.append(k)
+            values.append(v)
+    for k, v in session["placeholders"].items():
+        if k == "phone number":
+            keys.append(k)
+            values.append(v)
```

Comment:
Rather than looping through the dictionary again, would it make more sense to just grab it directly?  Also, we could add a bit of error handling if `phone number` doesn't exist for whatever reason, since it's required (unless we're already handling that elsewhere, and I suspect I'm forgetting that!).

```suggestion
    if "phone number" in session["placeholders"].keys():
        keys.append("phone number")
        values.append(session["placeholders"]["phone number"])
```